nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/remote/io/socket/ssl/SSLSocketChannel.java

```diff
@@ -582,7 +583,12 @@ public int read(final byte[] buffer, final int offset, final int len) throws IOE
                     continue;
                 }
                 case CLOSED:
-                    throw new IOException("Channel is closed");
+                    copied = copyFromAppDataBuffer(buffer, offset, len);
```

Comment:
It definitely appears to have... not sure how that happened. I do vaguely remember such an update but have no idea how it ended up in this PR. Will remove the code change from this PR.

Reply:
... and now I realize what it's here. While this is not technically related to this PR, it is a bug, and changes to the defaults in the JsonTreeReader have now exposed this in the TestListenTCPRecord unit test. So I fixed it here because otherwise that unit test will now fail. By fixing this here, the unit test no longer fails.